cpp/arcticdb/storage/azure/azure_mock_client.hpp

```diff
@@ -14,18 +14,11 @@
 
 #include <arcticdb/storage/storage_utils.hpp>
 #include <arcticdb/storage/azure/azure_client_wrapper.hpp>
+#include <arcticdb/storage/storage_mock_client.hpp>
 
 namespace arcticdb::storage::azure {
 
-enum class AzureOperation{
-    READ,
-    WRITE,
-    DELETE, // delete_blobs will fail for all objects if one of them triggers a delete failure
-    LIST,
-    EXISTS,
-};
-
-class MockAzureClient : public AzureClientWrapper {
+class MockAzureClient : public AzureClientWrapper, public MockStorageClient<std::string, Azure::Core::RequestFailedException> {
```

Comment:
Since AFAICT the MockStorageClient is only here for code reuse, not for runtime polymorphism, I think it would be cleaner to use composition over inheritance for it.

Here is a page on why to worry about multiple inheritance: https://stackoverflow.com/questions/406081/why-should-i-avoid-multiple-inheritance

And a more balanced one https://isocpp.org/wiki/faq/multiple-inheritance